stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -208,6 +214,97 @@ public void run() {
     });
   }
 
+  @ChromeDevtoolsMethod
+  public PerformSearchResponse performSearch(JsonRpcPeer peer, final JSONObject params) {
+    return mDOMProvider.postAndWait(new UncheckedCallable<PerformSearchResponse>() {
+      @Override
+      public PerformSearchResponse call() {
+        final PerformSearchRequest request = mObjectMapper.convertValue(
+            params,
+            PerformSearchRequest.class);
+
+        // Searches are case insensitive so convert
+        // the query to lower case for simpler comparisons.
+        final String lowerCaseQuery = request.query.toLowerCase();
+
+        final Object root = mShadowDOM.getRootElement();
+
+        // Each search action has a unique ID so that
+        // it can be queried later.
+        final String id = UUID.randomUUID().toString();
+
+        final ArrayList<Integer> results = new ArrayList<>();
+
+        findMatches(lowerCaseQuery, root, results);
+
+        mSearchResults.put(id, results);
+
+        final PerformSearchResponse response = new PerformSearchResponse();
+        response.searchId = id;
+        response.resultCount = results.size();
+
+        return response;
+      }
+    });
+  }
+
+  private void findMatches(String query, Object root, ArrayList<Integer> resultIds) {
```

Comment:
Prefer calling it 'element' instead of 'root'. We need to ensure we include the word 'element' because we don't really have another way to express in Java that this needs to be "an element" (and element means something very specific in this corner of the code) (and this also provides consistency with our other code that works with "elements"). Since this is a recursive method, calling it 'root' might be a bit misleading too. The `element` in the loop below can be called `childElement`.

Also prefer having the `resultIds` be of type `List<Integer>` since it's the convention in Java to strip away concrete types in favor of interfaces, especially across boundaries.